src/thread_task.rs

```diff
@@ -197,15 +197,19 @@ impl Rav1dTasks {
             let new_start = self.len();
             {
                 let mut pending_tasks = self.pending_tasks.lock();
+                if pending_tasks.len() == 0 {
+                    // A spurious merge.
+                    // We don't synchronise the two parts of `add_pending` so this is always possible.
```

Comment:
I'm not sure what this means, although I haven't looked at this code as much as you have lately.

Reply:
I'll include some more explanation.